spring-boot-project/spring-boot-test-autoconfigure/src/test/java/org/springframework/boot/test/autoconfigure/web/reactive/webclient/ExampleWebExceptionHandler.java

```diff
@@ -27,9 +29,11 @@
  * Example {@link WebExceptionHandler} used with {@link WebFluxTest} tests.
  *
  * @author Madhura Bhave
+ * @author Ali Dehghani
  */
 @Component
-public class ExampleWebExceptionHandler implements WebExceptionHandler {
+@ConditionalOnProperty(name = "custom-error-handler.enable")
```

Comment:
@alimate Could you explain why this change is necessary?

Reply:
@mbhave Apparently, As part of the #13627, We support the `WebExceptionHandler`s in `WebFluxTest`s. As far as I know, this behavior is inconsistent (Please correct me if I'm wrong) with when we run Spring Boot applications manually. In the latter scenario, we should register a bean of type `ErrorWebExxceptionHandler` to disable the default error handling and use a custom approach for error handling.

Basically, here I'm implementing the `ErrorWebExceptionHandler` interface to align error handling in automatic and manual testing scenarios.

 Also, I'm registering this new handler conditionally to test both of the following:
 - When there no `ErrorWebExceptionHandler` registered, the default provided by the `ErrorWebFluxAutoConfiguration` should be used.
 - Otherwise, the default one should be discarded and our custom `ErrorWebExceptionHandler` should be used.

Reply:
Thanks for the explanation. I think we'd still want to keep the `WebExceptionHandler` to make sure that those get [picked up](https://github.com/spring-projects/spring-boot/blob/master/spring-boot-project/spring-boot-test-autoconfigure/src/main/java/org/springframework/boot/test/autoconfigure/web/reactive/WebFluxTypeExcludeFilter.java#L53) by `@WebFluxTest`. For testing that the `ErrorWebFluxAutoConfiguration` is imported, the test that you've added in `WebFluxTestAutoConfigurationIntegrationTests` should suffice. 

Let's see what @bclozel thinks.

Reply:
I've created a sample repo for this inconsistent behaviour
https://github.com/alimate/webflux-Inconsistency

Reply:
This is not a case of `ErrorWebExceptionHandler` vs. `WebExceptionHandler`, since `ErrorWebExceptionHandler` is extending `WebExceptionHandler` after all.

This is more about the ordering of the beans - by default Spring Boot configures the `ErrorWebExceptionHandler` instance at order `-1` to make sure to handle errors before the `WebExceptionHandler` provided by Spring WebFlux.

An application can still configure a `WebExceptionHandler` earlier to process errors before the handler provided by Spring Boot.

In short, I agree with @mbhave and the `@ConditionalOnProperty` change is not necessary.

Reply:
@bclozel Thanks. So instead of registering the `ExampleWebExceptionHandler` as a `Component`, we're better off registering it manually using a `@Bean` with different `Order`s, right?

Reply:
Anyway, I removed the `Component` and `ConditionalOnProperty` annotations from the `ExampleWebExceptionHandler`. 

Instead, register the exception handler using a `@Bean` with different priorities to test the fact that `WebExceptionHandler`s with appropriate `Order`s would be called before the default handler provided by the `ErrorWebFluxAutoConfiguration`.

What do you guys think? @bclozel @mbhave

Reply:
@alimate I don't think we need to change anything related to `ExampleWebExceptionHandler`. The fact that a `WebExceptionHandler` can be configured with a higher precedence than the one provided by Spring Boot doesn't need to be tested in a test for `@WebFluxTest`. This issue is about ensuring that `ErrorWebFluxAutoConfiguration` is imported by `@WebFluxTest` and I think the only tests you need to change for that are `WebFluxTestAutoConfigurationIntegrationTests`.

Reply:
@mbhave OK. I'm reverting back some of the changes 👍